### ISSUE TITLE
Everything is inverted (mirrored)

### ISSUE THREAD

**Issue by julo42** [r0x0r/pywebview#1131]
### Specification
- pywebview version: 3.3.5+dfsg-1
- operating system: Debian Bullseye
- web renderer: GTK

### Description

I'm trying to start a webview in fullscreen to display an HTML page on the embedded screen. So, I used the following Python script to simply display Google's home page:

```
import webview

if __name__ == '__main__':
    webview.create_window('Test', 'http://www.google.com', fullscreen=True)
    webview.start()
```

When I run this script, the webview launches successfully, and opens Google's web page as expected. However, all the rendering is vertically inverted: (French) text goes from bottom to top, and the same goes for images.

### Practicalities

- NO I am not willing to work on this issue myself.

- NO I am not prepared to support this issue financially.

**Comment by julo42** [comment]
![screenshot](https://github.com/r0x0r/pywebview/assets/11225568/d2fddbaf-ed15-41e7-aac9-7911b734f4f5)

**Comment by r0x0r** [comment]
I have never encountered this problem. Does it work with QT?

**Comment by julo42** [comment]
I'm sorry, but I always get an error when trying with gui='qt':

qt.qpa.xcb: could not connect to display 
qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: eglfs, linuxfb, minimal, minimalegl, offscreen, vnc, wayland-egl, wayland, wayland-xcomposite-egl, wayland-xcomposite-glx, xcb.

**Comment by julo42** [comment]
Just to make things clear, I'm using a Variscite VAR-SOM-MX8M-PLUS board, which starts weston as root on boot. And, in this context, I can only start apps as root, with or without xwayland.

**Comment by r0x0r** [comment]
I am not familiar with this board, but judging my experience with installing QT/GTK dependencies on Raspberry PI, it might be the same case. Here is an old issue regarding RPI https://github.com/r0x0r/pywebview/issues/523

**Comment by julo42** [comment]
I don't see how this issue is related to the one you linked.

However, I managed to start weston as as non-root user, and, so, I could start my script using the QT GUI (as root I would get the "Running as root without --no-sandbox is not supported." error).

Now, with the QT backend and XWayland, all I get is a black screen and a lot of errors:

user@imx8mp-var-dart:~$ DISPLAY=:0.0 python3 test.py 
Unrecognized OpenGL version
Unrecognized OpenGL version
Unrecognized OpenGL version
Unrecognized OpenGL version
QGLXContext: Failed to create dummy context
/usr/lib/aarch64-linux-gnu/qt5/libexec/QtWebEngineProcess: /usr/lib/aarch64-linux-gnu/vivante/libOpenCL.so.1: no version information available (required by /lib/aarch64-linux-gnu/libavutil.so.56)
/usr/lib/aarch64-linux-gnu/qt5/libexec/QtWebEngineProcess: /usr/lib/aarch64-linux-gnu/vivante/libOpenCL.so.1: no version information available (required by /lib/aarch64-linux-gnu/libavutil.so.56)
/usr/lib/aarch64-linux-gnu/qt5/libexec/QtWebEngineProcess: /usr/lib/aarch64-linux-gnu/vivante/libOpenCL.so.1: no version information available (required by /lib/aarch64-linux-gnu/libavutil.so.56)
/usr/lib/aarch64-linux-gnu/qt5/libexec/QtWebEngineProcess: /usr/lib/aarch64-linux-gnu/vivante/libOpenCL.so.1: no version information available (required by /lib/aarch64-linux-gnu/libavutil.so.56)
Unrecognized OpenGL version
Unrecognized OpenGL version
QQuickWidget: Failed to make context current
QQuickWidget::resizeEvent() no OpenGL context
QQuickWidget: Failed to make context current
QQuickWidget: Attempted to render scene with no context
Unrecognized OpenGL version
Unrecognized OpenGL version
Unrecognized OpenGL version
Unrecognized OpenGL version
qt.qpa.backingstore: composeAndFlush: makeCurrent() failed
qt.qpa.backingstore: composeAndFlush: makeCurrent() failed
[1185:1185:0519/085905.438237:ERROR:gl_context_glx.cc(232)] Couldn't make context current with X drawable.
[1185:1185:0519/085905.438312:ERROR:gpu_info_collector.cc(85)] gl::GLContext::MakeCurrent() failed
[1185:1185:0519/085905.438373:ERROR:gpu_info_collector.cc(212)] Could not create context for info collection.
[1185:1185:0519/085905.438901:ERROR:gl_context_glx.cc(232)] Couldn't make context current with X drawable.
[1185:1185:0519/085905.438970:ERROR:gpu_info_collector.cc(85)] gl::GLContext::MakeCurrent() failed
[1185:1185:0519/085905.439004:ERROR:gpu_info_collector.cc(212)] Could not create context for info collection.
[1185:1185:0519/085905.449262:ERROR:gl_context_glx.cc(232)] Couldn't make context current with X drawable.
[1185:1185:0519/085905.449322:ERROR:gpu_channel_manager.cc(714)] ContextResult::kTransientFailure, failed to make context current
[1185:1185:0519/085905.449356:ERROR:shared_image_stub.cc(418)] SharedImageStub: unable to create context
[1185:1185:0519/085905.449406:ERROR:gl_context_glx.cc(232)] Couldn't make context current with X drawable.
[1185:1185:0519/085905.449430:ERROR:gpu_channel_manager.cc(714)] ContextResult::kTransientFailure, failed to make context current
[1185:1185:0519/085905.449451:ERROR:shared_image_stub.cc(418)] SharedImageStub: unable to create context
[1185:1185:0519/085905.449482:ERROR:gpu_channel.cc(441)] GpuChannel: Failed to create SharedImageStub
[1185:1185:0519/085905.449716:ERROR:gl_context_glx.cc(232)] Couldn't make context current with X drawable.
[1185:1185:0519/085905.449767:ERROR:gpu_channel_manager.cc(714)] ContextResult::kTransientFailure, failed to make context current
[1185:1185:0519/085905.449796:ERROR:shared_image_stub.cc(418)] SharedImageStub: unable to create context
[1185:1185:0519/085905.449841:ERROR:gl_context_glx.cc(232)] Couldn't make context current with X drawable.
[1185:1185:0519/085905.449858:ERROR:gpu_channel_manager.cc(714)] ContextResult::kTransientFailure, failed to make context current
[1185:1185:0519/085905.449879:ERROR:shared_image_stub.cc(418)] SharedImageStub: unable to create context
[1185:1185:0519/085905.449907:ERROR:gpu_channel.cc(441)] GpuChannel: Failed to create SharedImageStub
/usr/lib/aarch64-linux-gnu/qt5/libexec/QtWebEngineProcess: /usr/lib/aarch64-linux-gnu/vivante/libOpenCL.so.1: no version information available (required by /lib/aarch64-linux-gnu/libavutil.so.56)
/usr/lib/aarch64-linux-gnu/qt5/libexec/QtWebEngineProcess: /usr/lib/aarch64-linux-gnu/vivante/libOpenCL.so.1: no version information available (required by /lib/aarch64-linux-gnu/libavutil.so.56)
QQuickWidget: Attempted to render scene with no context

In conclusion, I cannot answer your question about how it works with QT because it is simply not possible to start it with the QT backend.
